File: frontend/client/lib/models/restaurant.dart
import 'food.dart';

class Restaurant {
  final int restaurantId;
  final String name;
  final String? address;
  final String? description;
  final List<Food> foods;

  Restaurant({
    required this.restaurantId,
    required this.name,
    this.address,
    this.description,
    this.foods = const [],
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      restaurantId: json['restaurantId'],
      name: json['name'],
      address: json['address'],
      description: json['description'],
      foods: (json['foods'] as List<dynamic>?)
              ?.map((f) => Food.fromJson(f))
              .toList() ??
          [],
    );
  }
}

File: frontend_fix/client/lib/model/Restaurant.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace client.lib.model
{
    public class Restaurant
    {
        [JsonPropertyName("restaurantId")]
        public int RestaurantId { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;

        [JsonPropertyName("address")]
        public string? Address { get; set; }

        [JsonPropertyName("description")]
        public string? Description { get; set; }

        [JsonPropertyName("foods")]
        public List<Food> Foods { get; set; } = new();
    }
}
